lib/core/src/side_swap/api/mod.rs

```diff
@@ -0,0 +1,342 @@
+        let req = Request::StartSwapWeb(StartSwapWebRequest {
+            asset: AssetId::from_str(&asset_swap.asset_id)?,
+            price: asset_swap.exchange_rate,
+            send_amount: asset_swap.payer_amount_sat as i64,
+            recv_amount: asset_swap.receiver_amount_sat as i64,
+            send_bitcoins: true,
+        });
+        let request_id = self.request_handler.send(req).await?;
+        let start_res = match self.response_handler.recv(request_id).await? {
+            Response::StartSwapWeb(res) => Ok(res),
+            res => Err(Self::invalid_response(res)),
+        }?;
+
+        let change_addr = self.onchain_wallet.next_unused_change_address().await?;
+
+        let wallet_utxos = self
+            .onchain_wallet
+            .asset_utxos(&utils::lbtc_asset_id(self.config.network))
+            .await?;
+
+        ensure_sdk!(
+            !wallet_utxos.is_empty(),
+            PaymentError::InsufficientFunds.into()
+        );
+
+        let mut inputs = vec![];
+        let mut send_value = 0i64;
+        for wallet_utxo in wallet_utxos {
+            send_value += wallet_utxo.unblinded.value as i64;
+            inputs.push(Utxo {
+                txid: wallet_utxo.outpoint.txid,
+                vout: wallet_utxo.outpoint.vout,
+                asset: wallet_utxo.unblinded.asset,
+                asset_bf: wallet_utxo.unblinded.asset_bf,
+                value: wallet_utxo.unblinded.value,
+                value_bf: wallet_utxo.unblinded.value_bf,
+                redeem_script: None,
+            });
+            if send_value >= start_res.send_amount {
+                break;
+            }
+        }
+        ensure_sdk!(
+            send_value >= start_res.send_amount,
+            PaymentError::InsufficientFunds.into()
+        );
+
+        let body = serde_json::to_string(&HttpRequest::SwapStart(SwapStartRequest {
+            order_id: start_res.order_id,
+            inputs,
+            recv_addr: receiver_address.clone(),
+            change_addr: change_addr.clone(),
+            send_asset: start_res.send_asset,
+            send_amount: start_res.send_amount,
+            recv_asset: start_res.recv_asset,
+            recv_amount: start_res.recv_amount,
+        }))?;
+        let (raw_body, status) = self
+            .rest_client
+            .post(
+                &start_res.upload_url,
+                Some(HashMap::from([(
+                    "Content-Type".to_string(),
+                    "application/json".to_string(),
+                )])),
+                Some(body),
+            )
+            .await?;
+
+        if !reqwest::StatusCode::from_u16(status)?.is_success() {
+            bail!("Received error status code when executing swap: {status}, body: {raw_body}");
+        }
+
+        let HttpResponse::SwapStart(response) = serde_json::from_str(&raw_body)? else {
+            bail!("Expected `SwapStart` response from server, received: {raw_body}");
+        };
```

Comment:
It might be nice to add a json fn to the RestClient to handle this serialize/deserialize pattern similar to the [payjoin post_request fn](https://github.com/breez/breez-sdk-liquid/blob/f48f71fd72acad3cd82c775ff99091292eb45c79/lib/core/src/payjoin/side_swap.rs#L88-L104)

Just thinking out loud, not requesting it now